orchestration/dagster_orchestration/hca_orchestration/tests/contrib/test_argo_workflows.py

```diff
@@ -0,0 +1,180 @@
+
+from unittest.mock import MagicMock, patch, call
+from typing import Generator, TypeVar, Optional
+from collections.abc import Iterable
+
+from hca_orchestration.contrib.argo_workflows import ExtendedArgoWorkflow, generate_argo_archived_workflows_client, \
+    ArgoFetchListOperation, ArgoArchivedWorkflowsClient
+from argo.workflows.client.models import V1alpha1Workflow, V1ObjectMeta, V1alpha1WorkflowSpec, V1alpha1Arguments, \
+    V1alpha1Parameter, \
+    V1alpha1WorkflowStatus
+
+from hca_orchestration.tests.support.mock_workflows import mock_argo_workflow
+
+
+def extend_workflow(workflow: V1alpha1Workflow) -> ExtendedArgoWorkflow:
+    return ExtendedArgoWorkflow(workflow, argo_url='https://nonexistentsite.test', access_token='token')
+
+
+class ArgoArchivedWorkflowsClientTestCase(unittest.TestCase):
+    def setUp(self):
+        self.client = ArgoArchivedWorkflowsClient("https://zombo.com", "tokentokentoken")
+
+    def workflows_to_page_of_results(self, archived_workflows: [], next_page: Optional[str]):
+        results_page = MagicMock()
+        results_page.items = archived_workflows
+        results_page.metadata._continue = next_page
+        return results_page
+
+    def test__pull_paginated_results_api_function_no_continue_results_one_call(self):
+        list_containing_3_V1alpha1Workflows = [
+            mock_argo_workflow('import-hca-total-defg', 'abc123uid', 'Succeeded'),
+            mock_argo_workflow('import-hca-total-abcd', 'abc234uid', 'Supsneeded', params={
+                'data-repo-name': 'datarepo_dataset1'
+            }),
+            mock_argo_workflow('import-hca-total-cdef', 'abc345uid', 'Succeeded', params={
+                'data-repo-name': 'datarepo_dataset2'
+            }),
+        ]
+
+        archived_workflow_list = self.workflows_to_page_of_results(list_containing_3_V1alpha1Workflows, None)
+
+        with patch('argo.workflows.client.ArchivedWorkflowServiceApi.list_archived_workflows',
+                   return_value=archived_workflow_list) as mock_list_archived_workflows:
+            list_results = list(self.client._pull_paginated_results(mock_list_archived_workflows))
+            self.assertEqual(list_results, list_containing_3_V1alpha1Workflows)
+            mock_list_archived_workflows.assert_called_once()
+
+    def test__pull_paginated_results_api_function_one_continue_results_two_calls(self):
+        list_containing_3_V1alpha1Workflows = [
+            mock_argo_workflow('import-hca-total-defg', 'abc123uid', 'Succeeded'),
+            mock_argo_workflow('import-hca-total-abcd', 'abc234uid', 'Supsneeded', params={
+                'data-repo-name': 'datarepo_dataset1'
+            }),
+            mock_argo_workflow('import-hca-total-cdef', 'abc345uid', 'Succeeded', params={
+                'data-repo-name': 'datarepo_dataset2'
+            }),
+        ]
+
+        archived_workflow_list_3_workflows = self.workflows_to_page_of_results(list_containing_3_V1alpha1Workflows,
```

Comment:
let's re-examine the names of these mock result pages as well, something like `mocked_first_page_response` might work better here